services/chat.js

```diff
@@ -33,6 +34,16 @@ const chatService = {
         select: '_id firstName lastName photo professionalSummary'
       }
     ])
+  },
+  deleteChat: async (id, currentUser) => {
+    const item = await Chat.findById(id).exec()
+    const isChatMember = item.members.some((member) => member.user.equals(currentUser))
+
+    if (!isChatMember) {
+      throw createForbiddenError()
+    }
+
+    await Chat.findByIdAndRemove(id).exec()
```

Comment:
As I see, the chat will be removed for both users, but as we discussed, it should be removed only for one user

Reply:
> As I see, the chat will be removed for both users, but as we discussed, it should be removed only for one user

This is one of the endpoints that we need to implement to fully delete chat, when both user have deleted it. We also will implement endpoint to "delete" chat only for one user. You can see details here:
https://github.com/ita-social-projects/SpaceToStudy-Client/issues/1188#issuecomment-1762887502